src/pip/_internal/operations/prepare.py

```diff
@@ -197,19 +196,9 @@ def unpack_url(
     #
     # As further cleanup, _copy_source_tree and accompanying tests can
     # be removed.
+    #
+    # TODO when use-deprecated=out-of-tree-build is removed
     if link.is_existing_dir():
-        deprecated(
-            reason=(
-                "pip copied the source tree into a temporary directory "
-                "before building it. This is changing so that packages "
-                "are built in-place "
-                'within the original source tree ("in-tree build").'
-            ),
-            replacement=None,
-            gone_in="21.3",
-            feature_flag="in-tree-build",
-            issue=7555,
-        )
```

Comment:
Do we need to show a deprecation message when `--use-deprecated` is specified?

Reply:
There is a message in that case, it moved to req_command.py. I hesitated but in the end I think a deprecation message is useful to show gone_in.